imports/plugins/core/core/server/no-meteor/resolvers/Query/packageSettings.js

```diff
@@ -0,0 +1,35 @@
+import { decodeShopOpaqueId } from "@reactioncommerce/reaction-graphql-xforms/shop";
+import ReactionError from "/imports/utils/ReactionError";
+
+/**
+ * @name Query.packageSettings
+ * @method
+ * @memberof Shop/GraphQL
+ * @summary Gets the primary shop ID
+ * @param {Object} _ - unused
+ * @param {Object} args - an object of all arguments that were sent by the client
+ * @param {String} args.shopId - ID of shop to query
+ * @param {String} args.name - name of package
+ * @param {Object} context - an object containing the per-request state
+ * @return {Promise<String>} The shop ID based on the domain in ROOT_URL
+ */
+export default async function packageSettings(_, args, context) {
+  const { name, shopId: opaqueShopId } = args;
+  const shopId = decodeShopOpaqueId(opaqueShopId);
+
+  const { collections, userHasPermission } = context;
+  const { Packages } = collections;
+
+  if (userHasPermission(["admin"], shopId) === false) {
+    throw new ReactionError("access-denied", "User does not have permission to update Bronto settings");
+  }
```

Comment:
If we want to replicate what the publication currently does (pretty sure we do), we should not throw an error here. The `public` field of `settings` has config that is intended to be public. So something more like this below:

```js
let { settings } = await Packages.findOne({ name, shopId });
if (!userHasPermission(["admin"], shopId)) {
  settings = { public: settings.public };
}
```

Also should check to make sure that `findOne` doesn't return `null`. Should throw a not-found in that case.